### PR TITLE
Initial implementation of custom text formats

### PR DESCRIPTION
**Pull request by madalingorbanescu**
**Description**
This PR is going to add a new custom text format: Highlighted

**Types of changes**
What types of changes does your code introduce?
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/formats/highlighted/index.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		style: 'style',
+	},
+	edit( { isActive, value, onChange } ) {
+
+		const onToggle = () => {
+			onChange(
+				toggleFormat( value, {
+					type: name,
+					attributes: {
+						style: 'background-color: var(----novablocks-color-1, #000); color: var(--novablocks-dark-1, #FFF); padding: 3px 6px;',
```
Comment (razwan) [inline src/formats/highlighted/index.js:29]: We should probably only rely on the class name for default styling. Also... `----novablocks-color-1` has way too many dashes 😁

File: src/formats/highlighted/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		};
+
+		return (
+			<Fragment>
+				<RichTextToolbarButton
+					icon="star-filled"
+					title={ __( 'Highlighted', 'nova-blocks' ) }
+					onClick={ onToggle }
+					isActive={ isActive }
+				/> )
```
Comment (razwan) [inline src/formats/highlighted/index.js:42]: `)` is leftover and shows up in the editor

File: src/formats/formats.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { highlighted } from './highlighted';
+
+/**
+ * WordPress dependencies
+ */
+const { registerFormatType } = wp.richText;
+
+function registerNovaBlocksFormats() {
+	[
+		highlighted
```
Comment (razwan) [inline src/formats/formats.js:10]: Do we have a list of formats that we want to ship?

**Review by razwan — CHANGES_REQUESTED** [review]

File: src/formats/highlighted/index.js (hunk 2 above)
Comment (madalingorbanescu) [reply src/formats/highlighted/index.js:42]: Done. Thanks👍

File: src/formats/highlighted/index.js (hunk 1 above)
Comment (madalingorbanescu) [reply src/formats/highlighted/index.js]: Thank you for the suggestion. I have removed the inline CSS and I am now relying on ``novaloblocks-highlighted`` class.

File: src/formats/formats.js (hunk 3 above)
Comment (madalingorbanescu) [reply src/formats/formats.js]: For the moment we have only the Highlighted style. But I think that in the future we could add at least 1 more this is why I choose to write it like this.

**Comment by madalingorbanescu** [comment]
Hey @razwan,

I have made some changes and added the format library as a package. Please take a look and let me know if everything is ok.

Thank you.

**Comment by madalingorbanescu** [comment]
After the latest changes from the Gutenberg plugin, the highlighted style is not needed anymore, so I am going to close this PR.

See [this](https://github.com/WordPress/gutenberg/pull/34680).

**Pull request closed without merging**